src/gui/Src/Gui/DisassemblerGraphView.cpp

```diff
@@ -410,7 +408,7 @@ void DisassemblerGraphView::paintNormal(QPainter & p, QRect & viewportRect, int
 
             for(Instr & instr : block.block.instrs)
             {
-                for(auto & line : instr.text.lines)
+                for(auto line : instr.text.lines)

```

Comment:
Is the copy here intentional?

Reply:
Yes, it was intentional, otherwise it would permanently change the original text. But now when I think about, maybe it wasn't a good idea to do it in a painting loop function. 
The best I can come up with is to save highlighting tokens right before painting, and restore their original color afterwards. Something like this:

```c++
std::vector<std::tuple<int, QColor, QColor>> prevColors;
// highlight selected tokens
for (int i = 0; i < line.size(); i++)
{
    auto & rt = line[i];
    if(rt.text == mHighlightToken.text)
    {
        prevColors.push_back(std::make_tuple(i, rt.textColor, rt.textBackground));
        rt.textColor = mInstructionHighlightColor;
        rt.textBackground = mInstructionHighlightBackgroundColor;
    }
}
// paint
RichTextPainter::paintRichText(...);
// restore old colors
for (const auto & col: prevColors) {
    auto & rt = line[std::get<0>(col)];
    rt.textColor = std::get<1>(col);
    rt.textBackground = std::get<2>(col);
}
```

Except this isn't pretty, and I am not sure there is no better way. Anyway, just tell me what to do and I'll fix it

Reply:
Yeah, in the disassembly this isn't an issue because the tokens are thrown away very quickly. With the graph view they are (or at least that was the intention) only fetched once. I think the 'best' solution would be to duplicate the rich text and then when the user performs a highlighting action:

- Restore the original rich text
- Apply highlighting

That way there is no copying/hackery going on during the painting.

Reply:
If by duplication you meant simply adding another field to Text, then this should be it